### PR TITLE
56 add login methods

### LINKED ISSUE #56: Add login methods

### PR DESCRIPTION
**Pull request by bendsp**
This branch added SMS login functionality, the privy UI login and fixed bugs causing login failures and crashes on android.

Also removes unused dependencies and updates several

### CODE REVIEW

File: app/_layout.tsx [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
   DMSerifText_400Regular_Italic,
 } from '@expo-google-fonts/dm-serif-text';
-import { PrivyProvider } from '@privy-io/expo';
+
+import { Inter_400Regular, Inter_500Medium, Inter_600SemiBold } from '@expo-google-fonts/inter';
+
+import { PrivyProvider, PrivyElements } from '@privy-io/expo';
 import { Drawer } from 'expo-router/drawer';
 import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import { useFonts } from 'expo-font';
```
Comment (PaulThblt) [inline app/_layout.tsx:22]: Is there a reason why you are using `useFonts` from `expo-font` and not `expo-google-fonts/dm-sans` ?

File: app/_layout.tsx [hunk 2]
```diff
@@ -32,6 +35,9 @@ const Layout = () => {
     DMSans_500Medium_Italic,
     DMSans_700Bold,
     DMSans_700Bold_Italic,
+    Inter_400Regular,
+    Inter_500Medium,
+    Inter_600SemiBold,
```
Comment (PaulThblt) [inline app/_layout.tsx:40]: Also why are we loading new fonts?

File: components/DebugButton.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        console.log('\n---------------------Privy data---------------------');
+        console.log('Privy User:', privyUser);
+        const accessToken = await getAccessToken();
+        console.log('\nPrivy Access Token:', accessToken);
+    };
+
+    return (
+        <Button onPress={debug} className="bg-primary" title="Debug" />
+    )
+};
```
Comment (PaulThblt) [inline components/DebugButton.tsx]:
```suggestion
};

```

File: index.ts [hunk 4]
```diff
@@ -4,5 +4,23 @@
...
+// If 'atob' exists but has issues, override it
+const _atob = global.atob;
+
+global.atob = (val) => {
+  try {
+    return _atob(val);
+  } catch (e) {
+    return _atob(val + '=');
+  }
+};
```
Comment (PaulThblt) [inline index.ts:24]: Why is this necessary again?

Comment (PaulThblt) [reply index.ts:24]: Im guessing this is a fix for the android bug right, but what was not working properly?

File: app/_layout.tsx (hunk 1 above)
Comment (bendsp) [reply app/_layout.tsx:22]: Yes. Not all fonts used come from `expo-google-fonts/dm-sans`

File: app/_layout.tsx (hunk 2 above)
Comment (bendsp) [reply app/_layout.tsx:40]: These are the fonts used by [Privy UI elements](https://docs.privy.io/guide/expo/ui/).

File: index.ts (hunk 4 above)
Comment (bendsp) [reply index.ts:24]: This fix came from Josh from the privy development team.
> Hey there these issue[s] are related to each other and due to an underlying bug with base 64 encoding and decoding the the android Javascript runtime
> you can override the built in decode method to solve this issue

File: app/_layout.tsx (hunk 2 above)
Comment (PaulThblt) [reply app/_layout.tsx:40]: Ohh I see, and theres no way to switch them with the ones we are using?

File: app/_layout.tsx (hunk 1 above)
Comment (PaulThblt) [reply app/_layout.tsx:22]: I think that the `useFont` method is the same from all the packages
But it's cleaner that way, just wanted to know

File: index.ts (hunk 4 above)
Comment (PaulThblt) [reply index.ts:24]: Thanks Josh 🙏

File: app/_layout.tsx (hunk 2 above)
Comment (bendsp) [reply app/_layout.tsx:40]: To my knowledge, no.

I'd like to discuss the usage of these UIs tomorrow

**Review by PaulThblt — APPROVED** [review]
